### ISSUE TITLE
Beef up tests

### ISSUE THREAD

**Issue by ZNielsen** [ZatchBo/numconverter#9]
Currently, the tests only exercise the core converter function, but there's logic to format the input before that function is called that remains untouched by the tests.

_edit_
Thanks to @Baev1, we have much better test coverage on the formatting logic.  This ticket is now open for anyone to add any tests.

**Comment by CalliEve** [comment]
Hi! I would like to work on this if that's alright ^^

**Comment by ZNielsen** [comment]
Of course!  I'll assign it to you and you can jump right in.

**Comment by CalliEve** [comment]
Would it be okay for me to split some parts of the main function into separate functions for better/easier testing? mainly the "Sort out the optional indexed argument" and "Convert input number to base 10" sections

**Comment by ZNielsen** [comment]
Absolutely.  Those probably should have been separate functions to begin with.

**Comment by ZNielsen** [comment]
Realized there could always be more tests.  Leaving this open in case anyone else wants to come along and add some.
